acvm-repo/acir_field/src/field_element.rs

```diff
@@ -0,0 +1,544 @@
+    {
+        self.to_hex().serialize(serializer)
+    }
+}
+
+impl<'de, T: ark_ff::PrimeField> Deserialize<'de> for FieldElement<T> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s: Cow<'de, str> = Deserialize::deserialize(deserializer)?;
+        match Self::from_hex(&s) {
+            Some(value) => Ok(value),
+            None => Err(serde::de::Error::custom(format!("Invalid hex for FieldElement: {s}",))),
+        }
+    }
+}
+
+impl<F: PrimeField> From<u128> for FieldElement<F> {
+    fn from(a: u128) -> FieldElement<F> {
+        let result = match F::from_str(&a.to_string()) {
+            Ok(result) => result,
+            Err(_) => panic!("Cannot convert u128 as a string to a field element"),
+        };
+        FieldElement(result)
+    }
+}
+
+impl<F: PrimeField> From<usize> for FieldElement<F> {
+    fn from(a: usize) -> FieldElement<F> {
+        FieldElement::from(a as u128)
+    }
+}
+
+impl<F: PrimeField> From<bool> for FieldElement<F> {
+    fn from(boolean: bool) -> FieldElement<F> {
+        if boolean {
+            FieldElement::one()
+        } else {
+            FieldElement::zero()
+        }
+    }
+}
+
+impl<F: PrimeField> FieldElement<F> {
+    pub fn from_repr(field: F) -> Self {
+        Self(field)
+    }
+
+    // XXX: This method is used while this field element
+    // implementation is not generic.
+    pub fn into_repr(self) -> F {
+        self.0
+    }
+
+    fn is_negative(&self) -> bool {
+        self.neg().num_bits() < self.num_bits()
+    }
+
+    fn fits_in_u128(&self) -> bool {
+        self.num_bits() <= 128
+    }
+
+    /// Returns None, if the string is not a canonical
+    /// representation of a field element; less than the order
+    /// or if the hex string is invalid.
+    /// This method can be used for both hex and decimal representations.
+    pub fn try_from_str(input: &str) -> Option<FieldElement<F>> {
+        if input.contains('x') {
+            return FieldElement::from_hex(input);
+        }
+
+        let fr = F::from_str(input).ok()?;
+        Some(FieldElement(fr))
+    }
+
+    // mask_to methods will not remove any bytes from the field
+    // they are simply zeroed out
+    // Whereas truncate_to will remove those bits and make the byte array smaller
+    fn mask_to_be_bytes(&self, num_bits: u32) -> Vec<u8> {
+        let mut bytes = self.to_be_bytes();
+        mask_vector_le(&mut bytes, num_bits as usize);
+        bytes
+    }
+
+    fn bits(&self) -> Vec<bool> {
+        fn byte_to_bit(byte: u8) -> Vec<bool> {
+            let mut bits = Vec::with_capacity(8);
+            for index in (0..=7).rev() {
+                bits.push((byte & (1 << index)) >> index == 1);
+            }
+            bits
+        }
+
+        let bytes = self.to_be_bytes();
+        let mut bits = Vec::with_capacity(bytes.len() * 8);
+        for byte in bytes {
+            let _bits = byte_to_bit(byte);
```

Comment:
The underscore in `_bits` makes it look like an unused variable pattern. Perhaps rename to `byte_bits`?